String/ValidateUrl.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * @function ValidateURL
+ * @description validate the URL.
+ * @param {String} url - The input URL string
+ * @return {Boolean}
+ * @see [ValidateURL](https://www.w3resource.com/javascript-exercises/javascript-regexp-exercise-9.php)
+ */
+const validateURL = (url) => {
+  const URL_PATTERN = /^(?:(?:https?|ftp):\/\/)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,5})?(?:\/\S*)?$/
```

Comment:
The regular expression is very inefficient. Please try to find other implementations of this functionality.

Reply:
For example:
```js
/(https?:\/\/(?:www\.|(?!www))[^\s.]+\.[^\s]{2,}|www\.[^\s]+\.[^\s]{2,})/gi
```

Reply:
update the regex pattern